src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -0,0 +1,118 @@
+(ns puppetlabs.trapperkeeper.services.authorization.authorization-core
+  (:require [clojure.string :as str]
+            [puppetlabs.kitchensink.core :as ks]
+            [schema.core :as schema]
+            [puppetlabs.trapperkeeper.authorization.rules :as rules]
+            [puppetlabs.trapperkeeper.authorization.acl :as acl])
+  (:import (java.util.regex PatternSyntaxException)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Constants
+
+(def required-keys
+  "Keys required in an auth rule map."
+  [:path :type])
+
+(def required-or-key
+  "At least one of these keys is required in an auth rule map."
+  [:deny :allow])
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(defn pprint-rule
+  [rule]
+  (str/trim (ks/pprint-to-string rule)))
+
+(schema/defn transform-config-rule-to-acl :- acl/ACL
+  [{:keys [allow deny]}]
+  (let [allow (if (string? allow) [allow] allow)
+        deny (if (string? deny) [deny] deny)]
+    (let [allow-acl (reduce #(acl/allow %1 %2) acl/empty-acl allow)
+          full-acl (reduce #(acl/deny %1 %2) allow-acl deny)]
+      full-acl)))
+
+(schema/defn transform-config-rule :- rules/Rule
```

Comment:
It feels to me like this function is reimplementing a lot of the functionality that already exists in tk-authorization for creating a rule.  What if instead we were to create a function similar to `config->rule` (https://github.com/puppetlabs/trapperkeeper-authorization/blob/1ac74458d6ee857f0fdcbf198584f635ed0ed1bd/src/puppetlabs/trapperkeeper/authorization/file/config.clj#L66-L74) which just starts from a config as it would be pulled from the TK service instead of a HOCON config object - reusing backing functions like `build-rule` and `add-acl`.  I'm thinking if we did that, we could get rid of the logic that builds up the rule map, including the entire `transform-config-rule-to-acl` function.  Does that make sense?